trezor/trezor-suite
# Tor external process - Allows Trezor Suite use Tor from Tails OS

## Description

This PR allows to run Trezor Suite with external Tor, in order to
* allow it to work in operating systems like Tails. 
* allow custom configuration for bridges that are not develop into Tor in Trezor Suite so use can configure a custom one for their needs

Like every great power it comes with great responsibility, so when running Tor External user could do something wrong that ruins their privacy so user has to know what he is doing when using this feature.

- [feat(node-utils): checkSocks5Proxy util](https://github.com/trezor/trezor-suite/pull/15642/commits/1f99643719e092c0afc52d598537aa58f5e57a7a) - new utility function `checkSocks5Proxy` to make sure there is a socks5 proxy running in the port for External Tor.
- [test(node-utils): checkSocks5Proxy tests](https://github.com/trezor/trezor-suite/pull/15642/commits/9730f57386010d6f218859eb8a9755525c2dd006) - Testing new utility function `checkSocks5Proxy`
- [feat(request-manager): support external Tor](https://github.com/trezor/trezor-suite/pull/15642/commits/1a40c65bef35a5b4914d5e37cd82664a3f85e414) - refactor Tor controller and create new controller for external Tor.
- [feat(suite-desktop-core): support external Tor](https://github.com/trezor/trezor-suite/pull/15642/commits/9a9eef2a42aef75cdc6a2a59da34113848089dec) - create new `TorExternalProcess` to allow handle external Tor and update the Tor module so it can change from bundled Tor to External Tor.
- [feat(suite-desktop-api): support external Tor](https://github.com/trezor/trezor-suite/pull/15642/commits/0670fffb9ad01784e7baca1e78c9b223dc481475) -  new type in TorSettings `useExternalTor`
- [feat(suite): external-tor experimental feature](https://github.com/trezor/trezor-suite/pull/15642/commits/0628e5179e6f35f373a3b1f16ccffb794244a858) - Add `External Tor` to experimental features in Suite.
- Removing snowflake experimental feature in favor of external Tor since with it the user can configure whatever Tor bridge they want.
## How to test it:

1. Enable Tor external experimental feature in Trezor Suite.
2. Run Tor daemon in port `9050` - Not necessary in Tails since it is running by default - but you need to make sure the circuit are established - Tails will ask you for it when starting.
3. Enable Tor in Trezor Suite.

When running and doing discover with a device connected you should be able to make sure that all the connection is going throw Tor by running `watch -n 1 'lsof -i TCP | grep electron'` and checking that there is not connection going to the open internet. You should see TCP connection to `localhost:9050` and no connection to anything else that is not in `localhost:xxxx`.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/5819  After 2 years open. :tada:
